### PR TITLE
Bugfixes and improvements

### PR DESCRIPTION
**Pull request by LukaszPiskadlo**
- Update spring boot
- Update gradle wrapper
- Update deprecated JPA Specifications methods
- Remove custom runtime exception handling
- Add wildcard character for search parameters

### CODE REVIEW

File: src/main/java/pl/edu/zut/mad/schedule/search/ScheduleSpecificationFactory.java [hunk 1]
```diff
@@ -60,12 +59,10 @@ public ScheduleSpecificationFactory(ScheduleExceptionFactory exceptionFactory) {
...
-        if (optionalSpecification.isPresent()) {
-            return Specifications.where(optionalSpecification.get())
-                    .and(new GroupScheduleSpecification(groupIds));
-        } else {
-            return Specifications.where(new GroupScheduleSpecification(groupIds));
-        }
+        return optionalSpecification.map(scheduleSpecification ->
+                Specification.where(scheduleSpecification)
+                        .and(new GroupScheduleSpecification(groupIds)))
+                .orElseGet(() -> Specification.where(new GroupScheduleSpecification(groupIds)));
```
Comment (MarekMacko) [inline src/main/java/pl/edu/zut/mad/schedule/search/ScheduleSpecificationFactory.java]: Definitely hard to read 😢

File: src/main/java/pl/edu/zut/mad/schedule/model/inner/Schedule.java [hunk 2]
```diff
@@ -113,6 +110,94 @@ public String getDay() {
         return day.trim();
     }
 
+    public Integer getGroupId() {
+        return groupId;
+    }
```
Comment (MarekMacko) [inline src/main/java/pl/edu/zut/mad/schedule/model/inner/Schedule.java:115]: I think that this getter could be replaced with `@Getter` annotation, it shouldn't affected to other getters.

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: src/main/java/pl/edu/zut/mad/schedule/model/inner/Schedule.java (hunk 2 above)
Comment (LukaszPiskadlo) [reply src/main/java/pl/edu/zut/mad/schedule/model/inner/Schedule.java:115]: It is the only getter, that would be affected by this change. I thin using `@Getter` annotation it this case would reduce readability.

File: build.gradle [hunk 3]
```diff
@@ -1,42 +1,40 @@
...
-	compile 'mysql:mysql-connector-java'
-	compile 'org.projectlombok:lombok'
-	compile 'com.h2database:h2'
-	testCompile('org.springframework.boot:spring-boot-starter-test')
+    compile 'org.springframework.boot:spring-boot-starter-web'
+    compile 'org.springframework.boot:spring-boot-starter-data-jpa'
+    compile 'mysql:mysql-connector-java'
+    compile 'org.projectlombok:lombok'
+    compile 'com.h2database:h2'
+    compile "io.springfox:springfox-swagger2:2.7.0"
```
Comment (MarekMacko) [inline build.gradle]: One small thing, please change `"` to `'` 😄

**Review by MarekMacko — APPROVED** [review]

**Pull request merged**